### DISCUSSION TITLE
Failed to boot Windows from iSCSI target

### DISCUSSION THREAD

**Discussion by shukuo** [ipxe/ipxe#215] [Q&A]
I am trying to use iPXE to boot windows server 2019 from iSCSI sw storage but ipxe will return error 0x3d222083. My iSCSI target is use targetcli to create in Red Hat. I could use sanhook to install Windows on iSCSI target successful but could not use sanboot to boot to Windows. Verify other method to boot Windows with same target it is work but through iPXE it does not. The script could work for Linux OS but Windows OS doesn't work. Below is my iPXE scripts.

>Install script:
#!ipxe
dhcp
set iscsi-target iscsi:192.168.1.1::::iqn.1991-05.com.abc:target
set gateway 0.0.0.0
set keep-san 1
sanhook ${iscsi-target}
echo Start Windows 2019 installation on iSCSI
kernel http://192.168.1.1/httpboot/wimboot
initrd http://192.168.1.1/WINPE-2019/Boot/BCD         BCD
initrd http://192.168.1.1/WINPE-2019/Boot/boot.sdi    boot.sdi
initrd http://192.168.1.1/WINPE-2019/sources/boot.wim boot.wim
boot

>Boot script:
#!ipxe
dhcp
set iscsi-target iscsi:192.168.1.1::::iqn.1991-05.com.abc:target
set gateway 0.0.0.0
set initiator-iqn iqn.2015-02.com.abc:uefi
set keep-san 1
sanboot ${iscsi-target}
boot

**Comment by NiKiZe** [comment]
If iPXE gives you an error, it will also give you an URL.
Post the full error message, including the URL.

**Reply by shukuo** [reply]
Here is the full error message:

Registered SAN device 0x80
Booting from SAN device 0x80
Boot from SAN device 0x80 failed: Error 0x3d222083 (http://ipxe.org/3d222083)
Unregistered SAN device 0x80
Could not boot image: Error 0x3d222083 (http://ipxe.org/3d222083)

**Reply by NiKiZe** [reply]
Good, could you verify your script? Try to copy the sanboot line and add an echo to see what you have.

Also your boot line will never run

**Reply by shukuo** [reply]
Yes, as you said it does not run the boot line

![image](https://user-images.githubusercontent.com/37550302/105469300-fc41f280-5cd2-11eb-8b6e-4cc3e7d11557.png)

**Reply by shukuo** [reply]
So, do you have any idea what may cause Boot from SAN device 0x80 failed?

**Reply by mcb30** [reply]
Given the specific error that you are seeing, it is most likely that your iSCSI disk is empty (or does not contain any EFI boot partition).
